### DISCUSSION TITLE
ControlNetModel from pre_trained does not work.

### DISCUSSION THREAD

**Discussion by nivedwho** [huggingface/diffusers#7416] [General]
I'm trying to load this ControlNet tile model https://huggingface.co/TTPlanet/TTPLanet_SDXL_Controlnet_Tile_Realistic_V1/ 
But it gives an error model_index.json is not found. Even from_single_file function also doesn't work for this.

**Comment by asomoza** [comment]
That controlnet is in diffusers format but he's not using the correct naming of the files, probably because he prefers to share it in a more "automatic1111" naming style as just a single file.

You can try asking him to rename the files o copy them to have the correct names to be able to use it in diffusers, they should be `diffusion_pytorch_model.safetensors` and `diffusion_pytorch_model.fp16.safetensors`. You can also download the files to a local directory and rename them yourself.

In the meantime I created a new repository with the correct name for the float16 version and you can load it like this:

```py
controlnet = ControlNetModel.from_pretrained(
    "OzzyGT/SDXL_Controlnet_Tile_Realistic", torch_dtype=torch.float16, variant="fp16"
)
```
